lib/widget/widgbase.h

```diff
@@ -209,6 +209,24 @@ class WIDGET
 	{
 		return dim.y();
 	}
+	int screenPosX() const
+	{
+		int screenX = x();
+		for (WIDGET *psParent = parent(); psParent != nullptr; psParent = psParent->parent())
```

Comment:
Using `auto` here would avoid having to change this code after the smart pointers refactor is merged.

We could also call the parent's `screenPosX` instead of looping through the parents' parents (if we don't care about the extra cost).